environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java

```diff
@@ -300,15 +300,23 @@ private boolean containsAnnotation(ClassInfo classInfo, DotName requiredAnnotati
                 // fallback to using reflection
                 Class<?> interfaceClass = loadClass(interfaceName.toString());
                 for (Method method : interfaceClass.getDeclaredMethods()) {
-                    if (method.isDefault() && method.isAnnotationPresent(requiredAnnotation)) {
+                    if (method.isDefault() && Reflections.containsAnnotations(method.getAnnotations(), requiredAnnotation)) {
                         return true;
                     }
                 }
                 continue;
             }
             for (MethodInfo method : interfaceInfo.methods()) {
-                if (isDefault(method) && method.hasAnnotation(requiredAnnotationName)) {
-                    return true;
+                if (isDefault(method)) {
+                    if (method.hasAnnotation(requiredAnnotationName)) {
+                        return true;
+                    }
+                    // Meta-annotations
+                    for (AnnotationInstance annotation : method.annotations()) {
```

Comment:
We are going to need a test case for this as well ;-)

Reply:
Eh, well you're so strict ;-)